assets/magnify.js

```diff
@@ -0,0 +1,43 @@
+// create a container and set the full-size image as its background
+const createOverlay = (image) => {
+  overlay = document.createElement('div');
+  overlay.setAttribute('class', 'image--full-size');
+  overlay.setAttribute('aria-hidden', 'true');
+  overlay.style.backgroundImage = `url('${image.src}')`;
+  return overlay;
+};
+
+const moveWithHover = (image, event, zoomRatio) => {
+  // calculate mouse position
+  const ratio = image.height / image.width;
+  const container = event.target.getBoundingClientRect();
+  const xPosition = event.clientX - container.left;
+  const yPosition = event.clientY - container.top;
+  const xPercent = `${xPosition / (overlay.clientWidth / 100)}%`;
+  const yPercent = `${yPosition / ((overlay.clientWidth * ratio) / 100)}%`;
+
+  // determine what to show in the frame
+  overlay.style.backgroundPosition = `${xPercent} ${yPercent}`;
+  overlay.style.backgroundSize = `${image.width * zoomRatio}px`;
+};
+
+const magnify = (image, zoomRatio) => {
+  // add full-size image on top of original
+  const overlay = createOverlay(image);
+  image.parentElement.insertBefore(overlay, image);
+
+  overlay.onclick = () => overlay.remove();
+  overlay.onmousemove = (event) => moveWithHover(image, event, zoomRatio);
+}
+
+const enableZoomOnHover = () => {
+  const images = document.querySelectorAll('.image--hover');
+  images && images.forEach(image => {
+    image.onclick = () => magnify(image, 2);
+  });
+}
+
+enableZoomOnHover();
+
+// also enable in theme editor
+document.addEventListener('shopify:section:load', () => enableZoomOnHover());
```

Comment:
Now sure if there's a better way to do this. Without it, the script only loads once in the editor, so when we change the setting from whatever is currently saved to "hover", we can't preview it.

Reply:
If I add the `shopify:section:load` event listener to `theme-editor.js`, how do I import the `magnify` function? I haven't seen a pattern for working with modules in our codebase.

Or do you mean move all of the js code in `magnify.js` into the `main-product` section with `{% javascript %}`, and using`request.design_mode` in there?